src/payment/strategies/chasepay/chasepay-payment-strategy.ts

```diff
@@ -0,0 +1,275 @@
+        this._methodId = methodId;
+
+        if (options.gatewayId === 'wepay') {
+            this._wepayRiskClient.initialize();
+        }
+
+        this._button = walletButton && document.getElementById(walletButton) || undefined;
+
+        if (this._button) {
+            this._button.addEventListener('click', this._handleWalletButtonClick);
+        }
+
+        return this._setChasePayConfiguration(
+            methodId,
+            onPaymentSelect,
+            this._storeLanguage
+        )
+        .then(() => super.initialize(options));
+    }
+
+    execute(orderRequest: OrderRequestBody, options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        const { payment, ...order } = orderRequest;
+        if (!payment || !options) {
+            throw new PaymentArgumentInvalidError(['payment']);
+        }
+
+        if (!this._paymentMethod || !this._paymentMethod.initializationData) {
+            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+        }
+
+        const { initializationData, config } = this._paymentMethod;
+
+        const riskToken = (payment.gatewayId === 'wepay') ? this._wepayRiskClient.getRiskToken() : undefined;
+        const paymentData = this._mapCardData(initializationData, riskToken);
+
+        return this._getPaymentData(payment.methodId,
+            initializationData.digitalSessionId,
+            config.testMode,
+            this._processPayment(payment, paymentData, order, options))
+            .then(() => super.initialize(options));
+    }
+
+    deinitialize(options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        if (this._button && options) {
+            this._button.removeEventListener('click', this._handleWalletButtonClick);
+        }
+
+        return super.deinitialize(options);
+    }
+
+    private _completeCheckout(payload: ChasePaySuccessPayload, methodId: string): Promise<any> {
+        const state = this._store.getState();
+        const method = state.paymentMethods.getPaymentMethod(methodId);
+        const requestId = method && method.initializationData && method.initializationData.merchantRequestId;
+        return this._setExternalCheckoutData(payload, requestId);
+    }
+
+    private _displayChasepayLightbox(digitalSessionId: string, testMode?: boolean): Promise<void> {
+        return this._chasePayScriptLoader.load(testMode)
+            .then(({ ChasePay }) => {
+                ChasePay.showLoadingAnimation();
+                ChasePay.startCheckout(digitalSessionId);
+            });
+    }
+
+    private _getPaymentData(methodId: string, digitalSessionId: string, testMode: boolean | undefined, onPaymentSelect: any): Promise<void> {
+        const state = this._store.getState();
+        const paymentMethod = state.paymentMethods.getPaymentMethod(methodId);
+
+        if (paymentMethod) {
+            return Promise.resolve(onPaymentSelect);
+        }
+
+        return this._setChasePayConfiguration(
```

Comment:
Also, you should only submit the order after the lightbox is closed (i.e.: after `COMPLETE_CHECKOUT` event) and reject the promise if the lightbox is cancelled.

Reply:
We use the `_setChasePayConfiguration` in the `initialize` to configure the callback `onPaymentSelect`. When we call it from `_getPaymentData` we configure it to use the callback to submit the payment once the user completes the login on the chasepay lightbox.

Reply:
Oh so you're passing in a different callback in. Doesn't that override the callback that's passed in the `initialize `method? We don't want that because the shopper can cancel the lightbox and click on the `walletButton` instead. Also you need to resolve the promise of `execute` after everything is done (after payment submission). What you have here doesn't wait for `_processPayment` to complete.

Reply:
Then if the user cancels the lightbox, we just need to add the configure on the cancel event.

Reply:
And you are right, we need to handle the wait until the user selects the card.